### DISCUSSION TITLE
The VSCP Daemon Phosphorus 15.0.0

### DISCUSSION THREAD

**Discussion by grodansparadis** [grodansparadis/vscp#264] [General]
It is my pleasure to announce that it is now time for release 15.0.0 Phosphorus of the VSCP & Friends package. From version 15 each component will be shipped separately and first out is the VSCP daemon, the server for the VSCP IoT system. 

The code for the full VSCP software framework has been fully rewritten and is now C++ standard based. There is no dependencies on external frameworks at all anymore. License is MIT. 

The goals for the software part of the project has changed. 

First: [node-red](https://nodered.org/)  and similar high level tools will be used for user interface and scheduling tasks. 

Secondly: MQTT is now a central part of VSCP and the VSCP daemon is dependent on it to work. VSCP is a good fit for MQTT. 

Third: much of the functionality that previously was built into the VSCP daemon is now available as drivers. This is true for the tcp/ip interface, the web server, the two websocket servers and the rest interface and some other functionality like remote variables that will follow. 

This release in fact make the VSCP daemon a low level interface to a MQTT based system. A low end driver can be hooked on to the VSCP daemon and the nodes handled by it (if any), or driver embedded functionality, can connect through the VSCP daemon and interface to a MQTT based system.

VSCP stands for the **Very Simple Control Protocol**, but is nowadays, with it's existence for more than two decades, more of an IoT/m2m framework. The most important part is still to make sure that a measurement, command etc, is interpreted at the source the same way as it was expected to be by the originator. 

The VSCP daemon can be downloaded [here](https://github.com/grodansparadis/vscp/releases). Packages for Debian based systems (also Raspberry Pi) is available. Windows install files will be available later.

If you want to know more about the VSCP framework the  [project home page](https://www.vscp.org)  is a good start. Most docs are [here](https://docs.vscp.org/ ) and especially for the VSCP daemon [here](https://grodansparadis.github.io/vscp/#/). If you want to discuss VSCP [this is the place](https://github.com/grodansparadis/vscp/discussions) to do so. If you find bugs or problems issue a ticket [here](https://github.com/grodansparadis/vscp/issues)

Of course I need to thank [my sponsors](https://github.com/sponsors/grodansparadis?preview=true) at this stage. Your support has been a great help.

Enjoy!
/Ake Hedman

-----

How to install

```
sudo apt install vscpd....
```
replace _vscpd...._ with one of the items below 

* **vscpd_15.0.0_debain.deb** is for Debian and Ubuntu and other Debian based systems. 
* **vscpd_15.0.0_raspberrypi.deb** is for Raspberry Pi systems.
* **vscpd_15.0.0_latest** Alternative for users working on the latest Ubuntu system. The file above should also work.

Build instructions etc is in the [documentation](https://grodansparadis.github.io/vscp/#/).

<hr /><em>This discussion was created from the release <a href='https://github.com/grodansparadis/vscp/releases/tag/v15.0.0'>The VSCP Daemon Phosphorus 15.0.0</a>.</em>

**Comment by datenheim** [comment]
Hello Ake,
the evolution VSCP has taken with this release is indeed great - great work - great leap!
I wondered quite some time if and when it will meet the MQTT universe because there is so much in common and much fits together... voila.
Thanks a lot for all the ~~fish~~ code.

**Reply by grodansparadis** [reply]
Yes, MQTT is added to the other transport protocols.   The old interface is still available but now as a driver.  Not much has been done about this lately because I currently work on the  vscp-works-qt project and it takes all my time. But hopefully this thing should go live soon now.
